### PR TITLE
Improve writeCodePointValue KDoc

### PR DESCRIPTION
**Pull request by fzhinkin**
Follow up to https://github.com/Kotlin/kotlinx-io/pull/308

### CODE REVIEW

**Review by shanshin — APPROVED** [review]

File: core/common/src/Utf8.kt [hunk 1]
```diff
@@ -130,6 +130,9 @@ internal fun String.utf8Size(startIndex: Int = 0, endIndex: Int = length): Long
  * Without such a conversion, data written to a [Sink] can not be converted back
  * to a string from which a surrogate pair was retrieved.
  *
+ * More specifically, all code points mapping to UTF-16 surrogates (`U+d800`..`U+dfff`)
+ * will be written as `?` characters (`U+0063`).
```
Comment (ilya-g) [inline core/common/src/Utf8.kt:134]: Is this JVM-only behavior or common one?

File: core/common/src/Utf8.kt [hunk 2]
```diff
@@ -130,6 +130,9 @@ internal fun String.utf8Size(startIndex: Int = 0, endIndex: Int = length): Long
...
+ *
  * @param codePoint the codePoint to be written.
  *
  * @throws IllegalStateException when the sink is closed.
```
Comment (ilya-g) [inline core/common/src/Utf8.kt:138]: Makes sense to document that IllegalArgumentException can be thrown.

File: core/common/src/Utf8.kt (hunk 1 above)
Comment (fzhinkin) [reply core/common/src/Utf8.kt:134]: Currently, it's a common one

File: core/common/src/Utf8.kt (hunk 2 above)
Comment (fzhinkin) [reply core/common/src/Utf8.kt:138]: https://github.com/Kotlin/kotlinx-io/pull/317

File: core/common/src/Utf8.kt (hunk 1 above)
Comment (ilya-g) [reply core/common/src/Utf8.kt:134]: This makes it inconsistent with `String.encodeToByteArray` where the replacement is unspecified, but different.

Comment (fzhinkin) [reply core/common/src/Utf8.kt:134]: `encodeToByteArray` call on a string with a low surrogate only results in a byte array containing `?`: https://pl.kotl.in/uFX3YULin

**Pull request merged**